File: frontend/src/types/stock.ts
export interface StockData {
  c: number;  // Current price
  d: number;  // Change
  dp: number; // Percent change
  h: number;  // High price of the day
  l: number;  // Low price of the day
  o: number;  // Open price of the day
  pc: number; // Previous close price
}

export interface CompanyProfile {
  name: string;
  ticker: string;
  exchange: string;
  industry: string;
  website: string;
  description: string;
  marketCap: number;
  employees: number;
  sector: string;
}

export interface StockSearchResult {
  symbol: string;
  name: string;
  exchange: string;
}

export interface HistoricalData {
  dates: string[];
  prices: {
    open: number[];
    high: number[];
    low: number[];
    close: number[];
    volume: number[];
  };
}

export interface PredictionData {
  dates: string[];
  predictions: number[];
  confidence: number[];
}

export interface PortfolioData {
  total_value: number;
  total_gain_loss: number;
  total_gain_loss_percent: number;
  last_updated: string;
  holdings: {
    symbol: string;
    shares: number;
    average_price: number;
    current_price: number;
    market_value: number;
    gain_loss: number;
    gain_loss_percent: number;
  }[];
}

export interface NewsItem {
  title: string;
  summary: string;
  url: string;
  source: string;
  published_date: string;
  sentiment_score: number;
  relevance_score: number;
} 

File: backend/app/models/stock.py
from pydantic import BaseModel, Field
from typing import List, Optional
from datetime import datetime

class StockData(BaseModel):
    symbol: str
    company_name: str
    current_price: float
    change: float
    change_percent: float
    volume: int
    market_cap: Optional[float]
    historical_data: List[dict]
    technical_indicators: dict

class StockPrediction(BaseModel):
    symbol: str
    current_price: float
    predicted_price: float
    confidence: float
    prediction_date: datetime
    prediction_interval: List[float]
    factors: dict

class NewsItem(BaseModel):
    title: str
    source: str
    url: str
    published_at: datetime
    summary: str
    sentiment_score: float
    relevance_score: float

class SentimentAnalysis(BaseModel):
    symbol: str
    overall_sentiment: float
    sentiment_trend: str
    news_sentiment: float
    social_sentiment: float
    technical_sentiment: float
    confidence: float
    last_updated: datetime
    sources: List[str] 